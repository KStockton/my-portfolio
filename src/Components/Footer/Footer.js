import React from 'react';
import { Link } from 'react-router-dom';
import formik from '../../assets/formik.png';
import awsLambda from '../../assets/awsLambda.png';
import awsgateway from '../../assets/awsgateway.png';
import { portfolioInfo } from '../../Util/Projects';
const shortid = require('shortid');

const Footer = () => {

  return (
    <div>
      <footer>
        <section className='one'>
          <h2>About Me</h2>
          <div>
            <p>
            {/* born and raised in denver, colorado. I have a background in chemistry andc */}
            {/* have been an entrpreneur sourcing private label items and selling them on Amazon.  */}
            
            i am a software engineer who loves the never-ending challenge between art, 
            analysis, creativity, and code. i am seeking a software engineering position on 
            an agile team that blends the never-ending challenge of complex problem solving 
            with open-minded curiosity.
            </p>
          </div>
        </section>
        <section>
          <Link to='/contact'>
            <div>
              <p>
                <i className="fas fa-envelope"></i>
              Contact Me
              </p>
            </div> 
          </Link>
        </section>
        <section>
          <h2>Activities</h2>
          <ul>
            { portfolioInfo.activities.map(activity =>
              <li key={shortid.generate()}>
                <i className={activity.classname}></i>
              </li>
            )}
          </ul>
        </section>
      </footer>
      <aside>
        <p>This page was built and deployed with:</p>
        <ul> 
          <li>
            <i className="fab fa-aws"></i>
          </li>   
          <li>
            <i className="fab fa-react"></i>
          </li>
          <li>
            <i className="fab fa-sass"></i>
          </li>
          <li>
            <img src={formik} alt='formik-logo'/>
          </li>
          <li>
            <img src={awsLambda} alt='aws lambda icon'/>
          </li>
          <li>
            <img src={awsgateway} alt='aws gateway api icon'/>
          </li>
        </ul>
      </aside>
    </div>
  );
};

export default Footer;