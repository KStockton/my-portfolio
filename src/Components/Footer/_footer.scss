footer {
  display: grid;
  grid-gap: 40px;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  margin: calc(10px + 2%);

  section:nth-child(1)  {
    position: relative;
    grid-column: 1;
    grid-row: 1 / 2;

    h2 {
      text-align: right;
      color: #ffffff;
      z-index: 1;
      padding: 3px;
      text-transform: capitalize;
      font-size: 20px;
      
      &::before {
        position: absolute;
        content: '';
        background-color: #ffa500;
        width: 10ch;
        height: 1.7rem;
        top: 0;
        bottom: 0;
        right: -1px;
        z-index: -1;
      }
    }

    div {
      position: relative;
      margin-top: 9px;

      blockquote {
        position: relative;
        color: black;
        text-align: left;
        font-size: 18px;
        font-family: 'PT', sans-serif;
        z-index: 2;
        font-style: italic;
        word-wrap: break-word;
        padding: 35px;
        
        &::before, &::after {
          display: block;
          font-size: 3em;
          color: #3333333d;
          position: absolute;
        }
        
        &::before {
          content: open-quote;
          top: 0px;
          left: 7px;
        }

        &::after {
          content: close-quote;
          bottom: -25px;
          right: 21px;
        }
      }


    }
  }

  section:nth-child(2) {
    background-color: #333;
    
    a {
      display: flex;
      align-items: center;
      justify-content: center;
      height: 100%;
      width: 100%;
      text-decoration: none;
      
      &:hover > * {
        cursor: pointer;
        border: 2px solid #ffd700;
      }
    }
    
    div {
      padding: 1rem;
      position: relative;
      border: 1px solid #ffffff;
      
      &::before, &::after {
        content: '';
        display: block;
        position: absolute;
        background: #333333;
        z-index: 1;
      }
      
      @include footer-corner-borders;
      
      p {
        position: relative;
        font-size: 24px;
        color: #ffffff;
        z-index: 2;
        cursor: pointer;
        
        i {
          padding-right: 8px;
        }
      }
    }
  }

  section:nth-child(3) {
    @include displayFlex(center, center, center, column);
    grid-column: 3;
    grid-row: 1;
    position: relative;
    border: 1px solid #333333;

    &::before, &::after {
      content: '';
      display: block;
      position: absolute;
      background: #ffffff;
      z-index: 1;
    }

    @include footer-corner-borders;

    h2 {
      text-align: right;
      text-transform: lowercase;
      z-index: 2;
    }

    ul {
      display: grid;
      grid-template-columns: repeat(4, 75px);
      grid-gap: 10px;
      margin: 10px;
      padding: 0px 0px 7px 0px;

      li {
        pointer-events: none;
        @include listItems;
        z-index: 2;
        background-color: #ffffff;
      }

      li:nth-child(1) {
        color: rgb(255, 165, 0);
      }

      li:nth-child(3) {
        color: rgb(255, 0, 0);
      }

      li:nth-child(5) {
        color: rgb(0, 191, 255);
      }

      li:nth-child(6) {
        color: rgb(164, 78, 74);
      }

      li:nth-child(7) {
        color: rgb(0, 128, 0);
      }

      li:nth-child(8) {
        color: rgb(164, 78, 74);
      }
    }
  }
}

aside {
  
  p {
    text-align: center;
    font-size: 18px;
  }

  ul {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(80px, 1fr));
    grid-gap: 10px;
    padding: 15px;
    align-items: center;
    justify-content: center;
    list-style-type: none;

      li:nth-child(4) img {
        height: 72px;
        width: 85px;
      }

      li:nth-child(5) img {
        height: 61px;
        width: 61px;
      }

      li:nth-child(6) img {
        height: 67px;
        width: 72px;
      }

      li {
        display: flex;
        align-items: center;
        justify-content: center;
        font-size: 30px;
        width: 100%;
        height: 100%;
      }
      
      li:nth-child(1) {
        color: #ffa500;
      }
      li:nth-child(2) {
        color: #62dfff;
      }
      li:nth-child(3) {
        color: #f80880;
      }
  }
}

@media only screen and (max-width: 1129px) {
  footer {
    grid-template-columns: 1fr;
    grid-template-rows: repeat(3, 1fr);
  
    section:nth-child(1){
      grid-column: 1;
      grid-row: 1;
    }

    section:nth-child(2){
      grid-column: 1;
      grid-row: 3;
    }

    section:nth-child(3) {
      grid-column: 1;
      grid-row: 2;
    }
  }
}

@media only screen and (max-width: 406px) {
  footer {
    grid-template-rows: 605px 150px 200px;

    section:nth-child(3) {
      
      h2 {
        font-size: 18px;
      }
      
      ul {
        grid-template-columns: repeat(4, 56px);
        margin: 0px;
        padding: 11px 0px 22px 0px;

        li {
          width: 43px;
          height: 37px;
          font-size: 24px;
        }
      }

    }
  }
}