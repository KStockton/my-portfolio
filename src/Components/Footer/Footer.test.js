import React from 'react';
import Footer from './Footer';
import { shallow } from 'enzyme';

describe('Footer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer/>);
  });

  xit('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
