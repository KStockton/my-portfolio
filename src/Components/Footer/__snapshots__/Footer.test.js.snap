// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match the snapshot 1`] = `
<div>
  <footer>
    <section
      className="one"
    >
      <h2>
        About Me
      </h2>
      <div>
        <blockquote>
          <span
            className="tab"
          >
            I am a Denver,
          </span>
           Colorado Native, and recent graduate from Turing School of Software and Design. I have a passion for software engineer because I enjoy the never-ending challenge between art, analysis, creativity, and code. 
          <br />
          <br />
           
          <span
            className="tab"
          >
             I am seeking
          </span>
           a software engineering position on an agile team that blends the never-ending challenge of complex problem solving with open-minded curiosity. My background ranges from a BS in Chemistry, UPS, Customer Service to Door to Door sells. For the past 8 years, I have been a mentor with kids in Big Brother Big Sister. Other times I enjoy hiking, movies, basketball, football, reading, to name a few. Thanks for visiting!
        </blockquote>
      </div>
    </section>
    <section>
      <Link
        to="/contact"
      >
        <div>
          <p>
            <i
              className="fas fa-envelope"
            />
            Contact Me
          </p>
        </div>
      </Link>
    </section>
    <section>
      <h2>
        Activities
      </h2>
      <ul>
        <li
          key="DFgtgRBAW"
        >
          <i
            className="fas fa-basketball-ball"
          />
        </li>
        <li
          key="dc0UzXDcut"
        >
          <i
            className="fas fa-hands-helping"
          />
        </li>
        <li
          key="FcDU22L5v3"
        >
          <i
            className="fas fa-book"
          />
        </li>
        <li
          key="lSZOFxMig5"
        >
          <i
            className="fas fa-hiking"
          />
        </li>
        <li
          key="-qF3Lew17p"
        >
          <i
            className="fas fa-plane"
          />
        </li>
        <li
          key="-gAvTyHdA6"
        >
          <i
            className="fas fa-football-ball"
          />
        </li>
        <li
          key="0bRfadM6yx"
        >
          <i
            className="fas fa-puzzle-piece"
          />
        </li>
        <li
          key="Twaj5SnG5g"
        >
          <i
            className="fas fa-dog"
          />
        </li>
      </ul>
    </section>
  </footer>
  <aside>
    <p>
      This page was built and deployed with:
    </p>
    <ul>
      <li>
        <i
          className="fab fa-aws"
        />
      </li>
      <li>
        <i
          className="fab fa-react"
        />
      </li>
      <li>
        <i
          className="fab fa-sass"
        />
      </li>
      <li>
        <img
          alt="formik-logo"
          src="formik.png"
        />
      </li>
      <li>
        <img
          alt="aws lambda icon"
          src="awsLambda.png"
        />
      </li>
      <li>
        <img
          alt="aws gateway api icon"
          src="awsgateway.png"
        />
      </li>
    </ul>
  </aside>
</div>
`;
